### PR TITLE
Testing via travis-ci

### PR DESCRIPTION
**Pull request by HerrSpace**
In this MR I:
- switch to django migrations
- fix the few existing tests to work again
- run those tests via travis ci on push

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+addons:
+  postgresql: "9.6"
+language: python
+python:
+  - "3.4"
+  - "3.5"
+  - "3.6"
+install:
+  - pip install .
+  - pip install -r serveradmin/requirements.txt
```
Comment (hasegeli) [inline .travis.yml]: Shouldn't the mail project be installed later?

File: serveradmin/settings.example.py [hunk 2]
```diff
@@ -15,11 +15,13 @@
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'HOST': 'localhost',
         'NAME': 'serveradmin',
+        'USER': 'serveradmin',
+        'PASSWORD': 'serveradmin',
```
Comment (hasegeli) [inline serveradmin/settings.example.py]: Without `HOST` it defaults to the Unix socket connection.  Then the operating system user would be allowed to connection to the database with the same username without a password.

File: serveradmin/settings.example.py [hunk 3]
```diff
@@ -15,11 +15,13 @@
...
         'OPTIONS': {
             'connect_timeout': 1,
             'client_encoding': 'UTF8',
-            'options': '-c lock_timeout=60000',
```
Comment (hasegeli) [inline serveradmin/settings.example.py]: I agree this setting is not a very good idea, though I though it was before.  It would be nice to remove this with a separate commit.

File: serveradmin/settings.example.py [hunk 4]
```diff
@@ -189,7 +191,7 @@
             'filters': ['require_debug_false'],
             'level': 'INFO',
             'formatter': 'verbose',
-            'filename': '/var/log/serveradmin.log',
+            'filename': '/tmp/serveradmin.log',
```
Comment (hasegeli) [inline serveradmin/settings.example.py]: I would also keep this under `/var/log` if possible.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  - psql -U postgres -c 'SELECT version();'
+  - psql -c "create user serveradmin with password 'serveradmin';" -U postgres
+  - psql -c 'alter user serveradmin superuser;' -U postgres
+  - psql -c 'create database serveradmin;' -U postgres
+  - cp serveradmin/settings.example.py serveradmin/settings.py
+  - ./manage.py migrate
+script:
+  # This will use pythons standard unit test discovery feature.
+  # Name your tests like test*.py
+  - python3 -Wall ./manage.py test
```
Comment (hasegeli) [inline .travis.yml]: Unnecessary `./`.

Comment (hasegeli) [inline .travis.yml]: Have you check if `manage.py test` is not already enables `-Wall`?

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+python:
+  - "3.4"
+  - "3.5"
+  - "3.6"
+install:
+  - pip install .
+  - pip install -r serveradmin/requirements.txt
+before_script:
+  - psql --version
+  - psql -U postgres -c 'SELECT version();'
```
[the remark this answers is not in the document]
Comment (HerrSpace) [reply .travis.yml]: It's just nice actually know what specific version a travis run used, as postgres is installed and provided to us by travis.

File: .travis.yml (hunk 1 above)
Comment (HerrSpace) [reply .travis.yml]: Makes more sense, yeah.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  - psql -c "create user serveradmin with password 'serveradmin';" -U postgres
```
[the remark this answers is not in the document]
Comment (HerrSpace) [reply .travis.yml]: Fixed

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  - psql -c 'alter user serveradmin superuser;' -U postgres
```
[the remark this answers is not in the document]
Comment (HerrSpace) [reply .travis.yml]: Fixed

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  - psql -c 'create database serveradmin;' -U postgres
```
[the remark this answers is not in the document]
Comment (HerrSpace) [reply .travis.yml]: Fixed

File: .travis.yml (hunk 5 above)
Comment (HerrSpace) [reply .travis.yml]: Fixed

File: serveradmin/dataset/fixtures/test_dataset.json [hunk 10]
```diff
@@ -1,199 +1,211 @@
 [
```
[the remark this answers is not in the document]
Comment (HerrSpace) [reply serveradmin/dataset/fixtures/test_dataset.json]: Fixed

File: serveradmin/settings.example.py (hunk 2 above)
Comment (HerrSpace) [reply serveradmin/settings.example.py]: Fixed

File: serveradmin/settings.example.py (hunk 3 above)
Comment (HerrSpace) [reply serveradmin/settings.example.py]: Fixed

File: serveradmin/settings.example.py (hunk 4 above)
Comment (HerrSpace) [reply serveradmin/settings.example.py]: fixed

File: serveradmin/serverdb_extra/migrations/0003_attribute_value_constraints.py [hunk 11]
```diff
@@ -1,24 +0,0 @@
...
-            'ALTER TABLE servertype_attribute '
-            'ADD CONSTRAINT servertype_attribute_default_value_check '
-            "   CHECK (default_value != '')"
-        ),
-        migrations.RunSQL(
-            'ALTER TABLE server_string_attribute '
-            'ADD CONSTRAINT server_string_attribute_value_check '
-            "   CHECK (value != '')"
-        ),
-    ]
```
[the remark this answers is not in the document]
Comment (HerrSpace) [reply serveradmin/serverdb_extra/migrations/0003_attribute_value_constraints.py]: Hu, I didn't give them much thought, but we really should keep them.. Of course it would be much better if django gave us to define them on the models but for now I guess I'll put these back in.. https://code.djangoproject.com/ticket/11964

[the remark this answers is not in the document]
Comment (HerrSpace) [reply serveradmin/serverdb_extra/migrations/0003_attribute_value_constraints.py]: I added them back in.

File: .travis.yml (hunk 5 above)
Comment (HerrSpace) [reply .travis.yml]: Yes, I have.
